pkg/resource/cluster.go

```diff
@@ -203,13 +203,18 @@ func (cluster Cluster) GetCockroachDBImageName() string {
 	supportedImages := getSupportedCrdbImages()
 	if cluster.Spec().CockroachDBVersion != "" {
 		if version, ok := cluster.LookupSupportedVersion(cluster.Spec().CockroachDBVersion); ok {
```

Comment:
It's possible I'm still missing something, but this logic doesn't look right to me.  The intent is to allow someone to either use one of the supported images via the CockroachDBVersion, or allow them to install any cockroachdb version.  In the latter case, we need to validate the version of crdb in the container and compare it to the current cluster version to make sure it is either a patch release of the current feature release, or the next newest feature release.  In the case of a new feature release, we need to also validate that any previous upgrade has been finalized.  This appears to kick out any image that isn't hosting an explicitly supported version of crdb.

Reply:
@keith-mcclellan  the CockroachDBVersion field can use the version checker only if it finds the image in the env var list we add. On line 203 we load the env vars we support and if this field is set we find a match on the supported versions to retrieve the container  image that will be loaded in the version checker job.   If we do not find an image we do not run  version checker job, the cluster will not be installed and we'll get an error on the status saying that we do not support the version given in the api field, or maybe to set the image field directly in the CR. With the api field we cannot run any version because we do not have the registry(docker, redhat etc), only with the image field we can do this logic at this time without hardcoding the registry.